### PR TITLE
RegistrationContextProvider

### PR DESCRIPTION
**Pull request by manojleaton**
Fixes #BLUI-5116 .

#### Changes proposed in this Pull Request:

- Added RegistrationContextProvider
- Updated the example project to use RegistrationContextProvider.
- Added the docs.

#### Screenshots / Screen Recording (if applicable)
Screenshot 2024-02-09 at 5 55 19 PM

#### To Test:

- yarn start:example-ios/android
- Check the `RegistratonContextExample`.

#### Any specific feedback you are looking for?

- Test cases are failing, checking on that. Please review rest of the things. Thanks

### CODE REVIEW

File: login-workflow/docs/registration-workflow.md
Comment (surajeaton) [inline login-workflow/docs/registration-workflow.md:1]: RegistrationContext and RegistrationWorkflowContext are two different things. This doc is about registration workflow. Please revert the changes in here

File: login-workflow/src/contexts/RegistrationContext/provider.tsx [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+    }
+
+    useEffect(() => {
+        void i18n.changeLanguage(language);
+    }, [i18n, language]);
+
+    return (
+        <I18nextProvider i18n={i18n}>
+            <RegistrationContext.Provider value={registrationContextProps}>
+                <ErrorContext.Provider value={errorConfig!}>{children}</ErrorContext.Provider>
```
[the remark this answers is not in the document]
Comment (manojleaton) [reply login-workflow/src/contexts/RegistrationContext/provider.tsx:42]: If we change to `ErrorContextProvider` getting this lint error

Screenshot 2024-02-09 at 10 45 31 PM

**Comment by JeffGreiner-eaton** [comment]
I'm having some issue viewing reg. context. I selected nav link and was presented with EULA screen and found the check box because its not rendered right now for some reason and click next but since doing that the app freezes up and stuck here...Noticed the same on Android too....is ProjectRegistrationUIActions wired up?

image

File: login-workflow/docs/registration-workflow.md [hunk 2]
```diff
@@ -2,11 +2,69 @@
...
+
+Because this workflow package is back-end agnostic, you must provide an implementation for what happens when the user triggers certain behaviors in the UI.
+
+The example project includes a skeleton implementation of all required functions using mocks / sleeps. You will need to replace these implementations with appropriate calls to your APIs, caching the returned data, etc. depending on the requirements of your application.
+
+1. Create a `RegistrationUIActions.ts` file
+    - This file will handle the implementation of the registration related actions (such as create password, verify code, and create account).
+    - You can copy this file directly from the [example](../example/src/actions/RegistrationUIActions.tsx) project as a starting point and then update the implementation details if you choose.
+2. You might also want to copy over the `example/src/store` and `example/src/constants` folders, which provide a very basic mechanism for storing important data using LocalStorage
+    - You will want to switch this out for a more secure approach before going to production with your application.
```
Comment (JeffGreiner-eaton) [inline login-workflow/docs/registration-workflow.md:29]: are we able to add in the example usage with the route & RegistrationContextProvider

**Comment by manojleaton** [comment]
> I'm having some issue viewing reg. context. I selected nav link and was presented with EULA screen and found the check box because its not rendered right now for some reason and click next but since doing that the app freezes up and stuck here...Noticed the same on Android too....is ProjectRegistrationUIActions wired up?
> 
> image

It is working fine in my local and @ektaghag-eaton's local, didn't used ProjectRegistrationUIActions with demo screens.

**Comment by JeffGreiner-eaton** [comment]
> > I'm having some issue viewing reg. context. I selected nav link and was presented with EULA screen and found the check box because its not rendered right now for some reason and click next but since doing that the app freezes up and stuck here...Noticed the same on Android too....is ProjectRegistrationUIActions wired up?
> > image
> 
> Hi @JeffGreiner-eaton , could you please share screen recording ?

[

https://github.com/etn-ccis/blui-react-native-workflows/assets/119693939/fb720286-3d2c-45ba-a604-f82207844dc1

](url)

**Comment by JeffGreiner-eaton** [comment]
@ektaghag-eaton this is from fresh clone. It did change some but results are similar and this time I can navigate and click to find check...

https://github.com/etn-ccis/blui-react-native-workflows/assets/119693939/458182f4-f365-4059-892a-339b02afe0e9

**Comment by ektaghag-eaton** [comment]
> @ektaghag-eaton this is from fresh clone. It did change some but results are similar and this time I can navigate and click to find check...
> 
>  fresh-clone.mov

Hi @JeffGreiner-eaton, that checkbox issue is in EulaScreen. I didn't focus on UI while designing EulaScreen since it was just a demo screen.

**Review by surajeaton — APPROVED** [review]
Looks good.

**Review by ektaghag-eaton — APPROVED** [review]

**Comment by surajeaton** [comment]
> @ektaghag-eaton this is from fresh clone. It did change some but results are similar and this time I can navigate and click to find check...
> 
>  fresh-clone.mov

Seems like the issue is related to demo eula screen, we can fix that while creating eula full screen
